java/kotlin-extractor/BUILD.bazel

```diff
@@ -0,0 +1,190 @@
+        ),
+        # * extractor.name is different for each version, so we need to put it in different output dirs
+        # * in order to put it in `resources`, we need to define `resource_strip_prefix` to strip this version
+        # * `resource_strip_prefix` is unique per jar, so we must also put other resources under the same version prefix
+        genrule(
+            name = "resources-%s" % v,
+            srcs = [src for src, _ in _resources],
+            outs = [
+                "%s/com/github/codeql/extractor.name" % v,
+            ] + [
+                "%s/%s" % (v, tgt)
+                for _, tgt in _resources
+            ],
+            cmd = "\n".join([
+                "echo %s-%s > $(RULEDIR)/%s/com/github/codeql/extractor.name" % (_extractor_name_prefix, v, v),
+            ] + [
+                "cp $(execpath %s) $(RULEDIR)/%s/%s" % (src, v, tgt)
+                for src, tgt in _resources
+            ]),
+        ),
+        kt_jvm_library(
+            name = "%s-%s" % (_extractor_name_prefix, v),
+            srcs =
+                [":generated-dbscheme"] +
+                glob(
+                    [
+                        "src/**/*.kt",
+                        "src/**/*.java",
+                    ],
+                    exclude = ["src/main/kotlin/utils/versions/**"],
+                ) + get_compatilibity_sources(v, "src/main/kotlin/utils/versions"),
+            kotlinc_opts = ":kotlinc-options-%s" % v,
+            module_name = "codeql-kotlin-extractor",
+            # resource_strip_prefix is very nit-picky: the following makes it work from
+            # `codeql`, `@codeql_kotlin_embeddable` and `semmle-code`
+            resource_strip_prefix = (
+                ("../%s/" % repo_name() if repo_name() else "") +
+                ("%s/" % package_name() if package_name() else "") +
+                v
+            ),
+            resources = [
+                ":resources-%s" % v,
+            ],
+            visibility = ["//visibility:public"],
+            deps = [
+                "@kotlin-compiler%s-%s" % (
+                    "-embeddable" if _for_embeddable else "",
+                    v,
+                ),
+                "@kotlin-stdlib-%s" % v,
+            ],
+        ),
+        # if in main repository, alias the embeddable versions from the modified @codeql_kotlin_embeddable repo
+        alias(
+            name = "%s-embeddable-%s" % (_common_extractor_name_prefix, v),
+            actual = "@codeql_kotlin_embeddable//:%s-embeddable-%s" % (_common_extractor_name_prefix, v),
+            visibility = ["//visibility:public"],
+        ) if not _for_embeddable else None,
+    )
+    for v in VERSIONS
+]
+
+# default aliases, based on the kotlinc version installed on the host
+# * default version can be overridden with env variableCODEQL_KOTLIN_SINGLE_VERSION
+# * setting CODEQL_KOTLIN_SINGLE_VERSION_EMBEDDABLE=true overrides the default variant
```

Comment:
Can we point to the documentation elsewhere, instead of duplicating the magic bazel invocation here?